### PR TITLE
Tech/eslint use more rules

### PR DESCRIPTION
**Pull request by alschmut**
# Tech/eslint use more rules

part of #878 

## Description

- Fixes code to satisfy the rules:
  - `no-prototype-builtins`
  - `no-useless-escape`

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

- [ ] **All** requirements mentioned in the issue are implemented
- [ ] Does match the Code of Conduct and the Contribution file
- [ ] Task has its own **GitHub issue** (something it is solving)
  - [ ] Issue number is **included in the commit messages** for traceability
- [ ] **Update the README.md** with any changes/additions made
- [ ] **Update the CHANGELOG.md** with any changes/additions made
- [ ] **Enough test coverage to ensure that uncovered changes do not break functionality**
- [ ] **All tests pass**
- [ ] **Descriptive pull request text**, answering:
  - What problem/issue are you fixing?
  - What does this PR implement and how?
- [ ] **Assign your PR to someone** for a code review
  - This person _will be contacted **first**_ if a bug is introduced into `master`
- [ ] **Manual testing** did not fail

### CODE REVIEW

File: visualization/app/codeCharta/util/nodeDecorator.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ export class NodeDecorator {
 			let root = d3.hierarchy<CodeMapNode>(map)
 			root.leaves().forEach(node => {
 				metricData.forEach(metric => {
-					if (!node.data.attributes.hasOwnProperty(metric.name)) {
+					if (node.data.attributes[metric.name] === undefined) {
```
Comment (NearW) [inline visualization/app/codeCharta/util/nodeDecorator.ts:98]:
```suggestion
					if (!node.data.attributes[metric.name]) {
```

File: visualization/app/codeCharta/util/settingsHelper.ts [hunk 2]
```diff
@@ -7,9 +7,9 @@ export function convertToVectors(settings: RecursivePartial<Settings>) {
 
 	for (let key of Object.keys(settings)) {
 		if (_.isObject(settings[key])) {
-			const xExists = settings[key].hasOwnProperty("x")
-			const yExists = settings[key].hasOwnProperty("y")
-			const zExists = settings[key].hasOwnProperty("z")
+			const xExists = settings[key]["x"] !== undefined
+			const yExists = settings[key]["y"] !== undefined
+			const zExists = settings[key]["z"] !== undefined
```
Comment (NearW) [inline visualization/app/codeCharta/util/settingsHelper.ts:12]:
```suggestion
			const xExists = settings[key]["x"]
			const yExists = settings[key]["y"]
			const zExists = settings[key]["z"]
```

File: visualization/app/codeCharta/util/nodeDecorator.ts (hunk 1 above)
Comment (alschmut) [reply visualization/app/codeCharta/util/nodeDecorator.ts:98]: GIVEN `const metricValue =  0`

THEN `!metricValue` is `true` 

AND `metricValue === undefined` is `false`

Comment (alschmut) [reply visualization/app/codeCharta/util/nodeDecorator.ts:98]: The only other solution is using 
```
Object.prototype.hasOwnProperty(node.data.attributes, metric.name)
```

**Comment by NearW** [comment]
LGTM, will test it tomorrow and merge it then.

**Pull request merged**